Content.Shared/Delivery/DeliveryModifierSystem.cs

```diff
@@ -0,0 +1,30 @@
+using Robust.Shared.Random;
+
+namespace Content.Shared.Delivery;
+
+/// <summary>
+/// System responsible for managing multipliers and logic for different delivery modifiers.
+/// </summary>
+public sealed partial class DeliveryModifierSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<DeliveryRandomMultiplierComponent, MapInitEvent>(OnRandomMultiplierMapInit);
+        SubscribeLocalEvent<DeliveryRandomMultiplierComponent, GetDeliveryMultiplierEvent>(OnGetRandomMultiplier);
+    }
+
+    private void OnRandomMultiplierMapInit(Entity<DeliveryRandomMultiplierComponent> ent, ref MapInitEvent args)
+    {
+        ent.Comp.CurrentMultiplierOffset = _random.NextFloat(ent.Comp.MinMultiplierOffset, ent.Comp.MaxMultiplierOffset);
+        Dirty(ent);
+    }
+
+    private void OnGetRandomMultiplier(Entity<DeliveryRandomMultiplierComponent> ent, ref GetDeliveryMultiplierEvent args)
+    {
+        args.Multiplier += ent.Comp.CurrentMultiplierOffset;
```

Comment:
mixing additive and multiplicative modifiers is kinda suspect. It means that the internal handling order of these events determines some very notable things. 

Your event should probably have a base price, multipliers, and flat modifiers. That way your adjustment can be based off of the base price (and not the price after some arbitrary modifiers) and can apply things in a deterministic order.

With the current implementation, you couldn't set the multiplier to 0 to void the value, as there is a chance that an event handling it afterwards would increment the multiplier.